flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -200,226 +125,224 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
-                })
-                .catch(err => {
-                    if (err.status === 404 || err === NXApi.disabledError || err === NXApi.disconnectedError) {
-                        return;
-                    }
-                    console.log("TELEX MSG FETCH FAILED");
-                });
-        }, NXApi.updateRate);
+    clearMessageQueue() {
+        this.messageQueue = [];
     }
 
     onUpdate(_deltaTime) {
-        super.onUpdate(_deltaTime);
-
-        if (this.pageUpdate) {
-            this.pageUpdate();
-        }
-        if (SimVar.GetSimVarValue("L:FMC_UPDATE_CURRENT_PAGE", "number") === 1) {
-            SimVar.SetSimVarValue("L:FMC_UPDATE_CURRENT_PAGE", "number", 0).then();
-            if (this.refreshPageCallback) {
-                this.refreshPageCallback();
+        if (this._delayedPageSwitch) {
+            this._delayedPageSwitch.timeRemaining -= _deltaTime;
+            if (this._delayedPageSwitch.timeRemaining < 0) {
+                this.updateRequest = false;
+                this._delayedPageSwitch.callback(this.fmc, this, this._delayedPageSwitch.args);
+                this._delayedPageSwitch = null;
             }
         }
 
-        this.checkAocTimes();
+        // the MCDUs are refreshed at a regular interval
+        if (this.pageRefreshThrottler.canUpdate(_deltaTime) !== -1) {
+            this.requestOnsideUpdate();
+        }
 
-        this.updateMCDU();
+        // and also on request (e.g. data changed), with a minimum time between updates
+        if (this.minPageUpdateThrottler.canUpdate(_deltaTime) !== -1 && this.updateRequest) {
+            this.updateRequest = false;
+            if (this.pageRedrawCallback) {
+                const delayedPageSwitch = this._delayedPageSwitch;
+                this.pageRedrawCallback();
+                this._delayedPageSwitch = delayedPageSwitch;
+            }
+        }
     }
 
-    /* MCDU UPDATE */
+    setDelayedPageSwitch(time, callback, ...args) {
+        this._delayedPageSwitch = {
+            timeRemaining: time,
+            callback: callback,
+            args: args,
+        };
+    }
 
     /**
-     * Checks whether INIT page B is open and an engine is being started, if so:
-     * The INIT page B reverts to the FUEL PRED page 15 seconds after the first engine start and cannot be accessed after engine start.
+     * Updates both MCDU displays
      */
-    updateMCDU() {
-        if (this.isAnEngineOn()) {
-            if (!this.initB) {
-                this.initB = true;
-                setTimeout(() => {
-                    if (this.page.Current === this.page.InitPageB && this.isAnEngineOn()) {
-                        CDUFuelPredPage.ShowPage(this);
-                    }
-                }, 15000);
-            }
+    requestUpdate() {
+        this.requestOnsideUpdate();
+        this.requestOffsideUpdate();
+    }
```

Comment:
**suggestion**: Consider:
```suggestion
    requestUpdate() {
        this.fmc.requestUpdate();
    }
```

**thought**: I don't know if there are ever situations in the A320 where one can have three MCDUs. If so, you could consider putting things in an array. This _might_ already clean up code right now, as I see quite some duplication regarding executing the same piece of code for L or R, or both, by index or other means.

Reply:
I think there is an option to have a third at the rear of the pedestal like the A330... We'd have to find a lot of information before we could do anything about that though, as documentation mostly talks about onside and offside FMGC/MCDU/Nav Tuners/etc.